typhon/display.py

```diff
@@ -1,19 +1,35 @@
+from enum import Enum
 import logging
 import os.path
-from warnings import warn
 
-from pydm.widgets.drawing import PyDMDrawingImage
-from qtpy import uic
-from qtpy.QtCore import Qt
+from pydm import Display
+from qtpy.QtCore import Property, Slot, Q_ENUMS
+from qtpy.QtWidgets import QHBoxLayout
+
+from .utils import ui_dir, TyphonBase, clear_layout
+from .widgets import TyphonDesignerMixin
 
-from .func import FunctionPanel
-from .signal import SignalPanel
-from .utils import ui_dir, clean_attr, clean_name, TyphonBase, grab_kind
 
 logger = logging.getLogger(__name__)
 
 
-class TyphonDisplay(TyphonBase):
+class TemplateTypes:
+    """Types of Available Templates"""
+    embedded_screen = 0
+    detailed_screen = 1
+    engineering_screen = 2
+
+    @classmethod
+    def to_enum(cls):
+        # First let's remove the internals
+        entries = [(k, v) for k, v in cls.__dict__.items()
+                   if not k.startswith("__")
+                   and not callable(v)
+                   and not isinstance(v, staticmethod)]
+        return Enum('TemplateEnum', entries)
+
+
+class TyphonDisplay(TyphonBase, TyphonDesignerMixin, TemplateTypes):
```

Comment:
I don't think this needs to subclass `TemplateTypes`

Reply:
For stupid Qt it does. When this class gets saved in a `.ui` file it stores the parameters as `TyphonDisplay.Engineering` e.t.c

Reply:
That's awful